src/main/java/org/sciborgs1155/robot/Drive/RealDrive.java

```diff
@@ -0,0 +1,110 @@
+package org.sciborgs1155.robot.Drive;
+
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.ADIS16448_IMU;
+import java.util.Set;
+
+import org.sciborgs1155.lib.FaultLogger;
+import org.sciborgs1155.lib.SparkUtils;
+import org.sciborgs1155.lib.SparkUtils.Data;
+import org.sciborgs1155.lib.SparkUtils.Sensor;
+
+public class RealDrive implements DriveIO {
+
+  CANSparkMax frontLeft = new CANSparkMax(1, MotorType.kBrushless);
+  CANSparkMax rearLeft = new CANSparkMax(2, MotorType.kBrushless);
+
+  CANSparkMax frontRight = new CANSparkMax(3, MotorType.kBrushless);
+  CANSparkMax rearRight = new CANSparkMax(4, MotorType.kBrushless);
+
+  RelativeEncoder leftEncoder = frontLeft.getEncoder();
+  RelativeEncoder rightEncoder = frontRight.getEncoder();
+
+  ADIS16448_IMU gyro = new ADIS16448_IMU();
+
+  Pose2d pose = new Pose2d();
```

Comment:
This field doesn't really need to be here. Whenever you want to get your pose, use odometry.getPoseMeters() (or something like that). In updateState, just call odometry.update, no need to store it's return value anywhere

Reply:
roger roger